crates/sui-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -97,7 +96,14 @@ impl SharedObjectCongestionTracker {
         }
         let start_cost = self.compute_tx_start_at_cost(&shared_input_objects);
 
-        if start_cost + tx_cost <= max_accumulated_txn_cost_per_object_in_commit {
+        if self.allow_overage {
+            // In case overage is allowed, the cost of this transaction is irrelevant. We will
+            // always allow it to be scheduled as long as the budget is not already exceeded,
+            // and keep track of the debt later.
+            if start_cost <= self.max_accumulated_txn_cost_per_object_in_commit {
```

Comment:
Suggestion: use `<` instead of `<=` is more intuitive I think. If the max cost is 10, once object accumulated cost hits 10, we probably don't want to allow one more transaction.

Reply:
I agree that practically speaking, this doesn't matter much. But I think we probably don't want to allow overage in tx count mode, otherwise, we would see that say the limit is 10, the system allows 11 txn and then back to 10.

Reply:
If we don't want to allow overage in tx count mode, we can set the overage parameter to 0?